impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -492,32 +484,59 @@ private List<String> buildScramArguments() {
         }
     }
 
-    private void createScramUsersInZookeeper(KafkaConfig config, List<String> scramArguments) {
+    private void createScramUsersInZookeeper(Class<Server> kafkaServerClazz, List<String> scramArguments, KafkaConfig config) {
         if (!scramArguments.isEmpty()) {
             var uscrs = ScramUtils.getUserScramCredentialRecords(scramArguments);
-            ZKClientConfig zkClientConfig = KafkaServer.zkClientConfigFromKafkaConfig(config, false);
+            ZKClientConfig zkClientConfig = zkClientConfigFromKafkaConfig(kafkaServerClazz, config);
             try (var zkClient = createZkClient("invm-kafka-zookeeper-client", Time.SYSTEM, config, zkClientConfig)) {
-                var adminZkClient = new AdminZkClient(zkClient, Option.empty());
+                var adminZkClientClazz = Class.forName("kafka.zk.AdminZkClient");
+                var adminZkClient = ReflectionUtils.construct(adminZkClientClazz, zkClient, Option.empty()).orElseThrow();
                 var userEntityType = "users";
                 disableControllerCheck(zkClient);
                 uscrs.forEach(credentials -> {
-                    var userConfig = adminZkClient.fetchEntityConfig(userEntityType, credentials.name());
+                    var userConfig = fetchEntityConfig(adminZkClient, userEntityType, credentials.name());
                     var credentialsString = ScramCredentialUtils.credentialToString(ScramUtils.asScramCredential(credentials));
 
                     userConfig.setProperty(ScramMechanism.fromType(credentials.mechanism()).mechanismName(), credentialsString);
-                    adminZkClient.changeConfigs(userEntityType, credentials.name(), userConfig, false);
+                    changeConfigs(adminZkClient, userEntityType, credentials.name(), userConfig, false);
                 });
             }
+            catch (Exception e) {
+                throw new IllegalStateException("Failed to create scram users", e);
+            }
         }
     }
 
+    private void changeConfigs(Object adminZkClient, String userEntityType, String name, Properties userConfig, boolean isUserClientId) {
+        ReflectionUtils.invokeInstanceMethod(adminZkClient, "changeConfigs", userEntityType, name, userConfig, isUserClientId);
+    }
+
+    private Properties fetchEntityConfig(Object adminZkClient, String userEntityType, String name) {
+        return ReflectionUtils.invokeInstanceMethod(adminZkClient, "fetchEntityConfig", userEntityType, name);
+    }
+
+    private AutoCloseable createZkClient(String name, Time system, KafkaConfig config, ZKClientConfig zkClientConfig) {
```

Comment:
nit: do we really need to pass `name` and `time` down here? IntelliJ highlights that they are always invoked with the same args.